### DISCUSSION TITLE
Emulator command line arguments for automatically running compiled program

### DISCUSSION THREAD

**Discussion by bobbarnes1981** [slu4coder/Minimal-64x4-Home-Computer#8] [Ideas]
My process for writing code has the following steps:

1. write some code
2. compile code
3. open emulator
4. receive code in emulator
5. run
6. test
7. go to 1

This can be time consuming. I think it would be convenient if the emulator allowed passing a binary file on the command line so it could be run automatically. For example:

`.\Minimal64x4.exe prog.bin`

Then I could use a script to build and run the code something like this:

`.\asm.exe .\prog.asm > prog.bin && .\Minimal64x4.exe prog.bin`

**Comment by slu4coder** [comment]
That sounds like a good idea! Wouldn't it be even better if we could run the emulator and pass a *source* assembler file to it for automatic assembly and execution? Will think about it...

**Comment by slu4coder** [comment]
I've implemented full command line options into the emulator. As an example, we can now use the "l.." and "p.." commands of the spash menu as command line options:

Minimal64x4.exe lprgs/stars.asm p2000

Will load, assemble and run the source code stars.asm in /prgs. You will have to update the emulator, edit.asm and os.asm (or flash.bin) for this. Any feedback is appreciated.

**Reply by bobbarnes1981** [reply]
This is great, it makes it really quick to test my code.
